### PR TITLE
feat: add button in settings tab to clear database and all data from app

### LINKED ISSUE #137: add functionality to completely clear database
basically a 'hard reset' of the database

### PR DESCRIPTION
**Pull request by hagoptaminian**
closes #137

### CODE REVIEW

File: public/app/listeners/deleteAll.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  EXECUTIONS_COLLECTION,
+  ALGORITHMS_COLLECTION,
+  DATASETS_COLLECTION,
+  SCHEMAS_COLLECTION,
+} = require('../db');
+const { ERROR_GENERAL } = require('../../shared/errors');
+const { ALGORITHMS_FOLDER, DATASETS_FOLDER } = require('../config/paths.js');
+const { DELETE_ALL_ERROR, DELETE_ALL_SUCCESS } = require('../../shared/types');
+
+const deleteAll = (mainWindow, db) => () => {
```
Comment (pyphilia) [inline public/app/listeners/deleteAll.js:14]: Maybe `clearDatabase` would be clearer

**Review by pyphilia — APPROVED** [review]
Very nice! I just suggested to rename `deleteAll` to `clearDatabase` (which might change all other files), but everything else is great!

**Review by Painguin — APPROVED** [review]
Looks good and works great!

File: public/app/listeners/deleteAll.js (hunk 1 above)
Comment (hagoptaminian) [reply public/app/listeners/deleteAll.js:14]: 👍 makes sense

**Pull request merged**